processes/processes.go

```diff
@@ -23,85 +23,160 @@ import (
 
-	"github.com/intelsdi-x/snap/control/plugin"
-	"github.com/intelsdi-x/snap/control/plugin/cpolicy"
-	"github.com/intelsdi-x/snap/core"
-	"github.com/intelsdi-x/snap/core/serror"
-
-	"github.com/intelsdi-x/snap-plugin-utilities/config"
-	"github.com/intelsdi-x/snap-plugin-utilities/str"
+	"github.com/intelsdi-x/snap-plugin-lib-go/v1/plugin"
 )
 
 const (
-	//plugin name
-	pluginName = "processes"
+	// PluginName is name of plugin
+	PluginName = "processes"
+	// PluginVersion is version of plugin
+	PluginVersion = 8
+
 	//plugin vendor
 	pluginVendor = "intel"
 	// fs is proc filesystem
 	fs = "procfs"
-	//version of plugin
-	version = 7
+
+	// Namespace offsets
+	nsCategory  = 3
+	nsProcName  = 4
+	nsStateName = 4
+	nsPid       = 5
+	nsPsCount   = 5
+	nsPidMetric = 6
 )
 
 var (
 	metricNames = map[string]label{
 		"ps_vm": label{
+			category:    "pid",
 			description: "Virtual memory size in bytes",
 			unit:        "B",
 		},
 		"ps_rss": label{
+			category:    "pid",
 			description: "Resident Set Size: number of pages the process has in real memory",
 			unit:        "",
 		},
 		"ps_data": label{
+			category:    "pid",
 			description: "Size of data segments",
 			unit:        "B",
 		},
 		"ps_code": label{
+			category:    "pid",
 			description: "Size of text segment",
 			unit:        "B",
 		},
 		"ps_stacksize": label{
+			category:    "pid",
 			description: "Stack size",
 			unit:        "B",
 		},
 		"ps_cputime_user": label{
+			category:    "pid",
 			description: "Amount of time that this process has been scheduled in user mode",
 			unit:        "Jiff",
 		},
 		"ps_cputime_system": label{
+			category:    "pid",
 			description: "Amount of time that this process has been scheduled in kernel mode",
 			unit:        "Jiff",
 		},
 		"ps_pagefaults_min": label{
+			category:    "pid",
 			description: "The number of minor faults the process has made",
-			unit:        "",
+			unit:        "number",
 		},
 		"ps_pagefaults_maj": label{
+			category:    "pid",
 			description: "The number of major faults the process has made",
-			unit:        "",
+			unit:        "number",
 		},
 		"ps_disk_ops_syscr": label{
+			category:    "pid",
 			description: "Attempt to count the number of read I/O operations",
-			unit:        "",
+			unit:        "number",
 		},
 		"ps_disk_ops_syscw": label{
+			category:    "pid",
 			description: "Attempt to count the number of write I/O operations",
-			unit:        "",
+			unit:        "number",
 		},
 		"ps_disk_octets_rchar": label{
+			category:    "pid",
 			description: "The number of bytes which this task has caused to be read from storage",
 			unit:        "B",
 		},
 		"ps_disk_octets_wchar": label{
+			category:    "pid",
 			description: "The number of bytes which this task has caused, or shall cause to be written to disk",
 			unit:        "B",
 		},
+		"ps_cmdline": label{
+			category:    "pid",
+			description: "Process command line with arguments",
+			unit:        "string",
```

Comment:
skip a unit for this kind of metric - string is a data type, and there is no need to send this info in each series of this metric.